### PR TITLE
Generic "API Access", "Analyze via Odata" buttons and modal

### PR DESCRIPTION
**Pull request by ktuite**
These buttons are used on the submission list page, but are also going to be used on the entity list page for a very similar purpose.

Screen Shot 2023-02-08 at 3 44 06 PM

This PR makes these components more generic
- OData URL specified as a prop, not computed from form
- Whether or not to disable the button and show a tooltip also being pre-computed and passed in as props

Note - I moved some of the files around but decided to keep the old 2017/2020 dates even though the files look "new".

### CODE REVIEW

**Review by matthew-white — COMMENTED** [review]
Good idea to split this into its own PR! :+1:

For `SubmissionAnalyze`, I'm thinking that we should hold off on moving that file. That component has a fair number of i18n messages, and I think we should avoid having them retranslated (see #722). However, you can still modify and rename the component even if you don't move the file. We do something similar with src/components/submission/decrypt.vue. That component's filename is decrypt.vue, but it is imported everywhere as `SubmissionDownload`, not `SubmissionDecrypt`. Once #722 is resolved, I'm planning to rename decrypt.vue to download.vue, and we could do something similar for submission/analyze.vue.

File: src/components/odata/data-access.vue [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <span id="odata-access">
+    <a class="btn btn-default"
+      href="https://odkcentral.docs.apiary.io/#reference/odata-endpoints"
```
[the remark this answers is not in the document]
Comment (ktuite) [reply src/components/odata/data-access.vue:15]: I think the same URL can be used -- there's useful information about OData in general at the top of this section of the docs, and then a "Form Service" section. The Entity OData info could go in a "Entity Service" section right below.

File: src/components/form/submissions.vue [hunk 2]
```diff
@@ -18,23 +18,25 @@ except according to the terms contained in the LICENSE file.
         <enketo-fill v-if="rendersEnketoFill" :form-version="form">
           <span class="icon-plus-circle"></span>{{ $t('action.createSubmission') }}
         </enketo-fill>
-        <submission-data-access :form-version="form"
+        <o-data-access :analyze-disabled="analyzeDisabled"
```
[the remark this answers is not in the document]
Comment (ktuite) [reply src/components/form/submissions.vue]: `OdataWhatever` and `odata-whatever` does seem a bit better than `ODataWhatever` at the expense of `o-data-whatever`. I'll change it!

[the remark this answers is not in the document]
Comment (ktuite) [reply src/components/form/submissions.vue]: Wait, you know what's throwing me off? It was `SubmissionDataAccess`, and I moved it from `submssion/data-access.vue` to `odata/data-access.vue`, so perhaps it should be `OdataDataAccess`? 😂

[the remark this answers is not in the document]
Comment (matthew-white) [reply src/components/form/submissions.vue]: Haha good catch! Either way sounds good to me. Either renaming the component to `OdataDataAccess` or renaming the file to odata/access.vue.

**Pull request merged**